### PR TITLE
#2187 Renaming Repeated Columns in dataframes

### PR DESCRIPTION
**Pull request by pdet**
In case a dataframe has multiple columns with the same name, subsequent columns are renamed adding _x where x is the repetition number.

### CODE REVIEW

File: tools/pythonpkg/src/vector_conversion.cpp [hunk 1]
```diff
@@ -313,6 +313,23 @@ void VectorConversion::BindPandas(py::handle df, vector<PandasColumnBindData> &b
 	if (py::len(df_columns) == 0 || py::len(df_types) == 0 || py::len(df_columns) != py::len(df_types)) {
 		throw std::runtime_error("Need a DataFrame with at least one column");
 	}
+
+	// check if names in pandas dataframe are unique
+	unordered_map<string, idx_t> pandas_column_names_map;
+	py::array column_attributes = df.attr("columns").attr("values");
+	for (idx_t col_idx = 0; col_idx < py::len(df_columns); col_idx++) {
+		pandas_column_names_map[py::str(df_columns[col_idx])]++;
```
Comment (Mytherin) [inline tools/pythonpkg/src/vector_conversion.cpp:321]: The performance here is unlikely to matter, but it seems somewhat wasteful to call `py::str(df_columns[col_idx])` so many times. Could we call it once and extract it into a `string` variable?

File: tools/pythonpkg/src/vector_conversion.cpp [hunk 2]
```diff
@@ -313,6 +313,23 @@ void VectorConversion::BindPandas(py::handle df, vector<PandasColumnBindData> &b
...
+		if (pandas_column_names_map[py::str(df_columns[col_idx])] > 1) {
+			// If the column name is repeated we start adding _x where x is the repetition number
+			string column_name = py::str(df_columns[col_idx]);
+			column_name += "_" + to_string(pandas_column_names_map[py::str(df_columns[col_idx])] - 1);
```
Comment (Mytherin) [inline tools/pythonpkg/src/vector_conversion.cpp:325]: What if there is a conflict (e.g. you try to rename `a` to `a_1` but `a_1` already exists as a column?)

Comment (pdet) [reply tools/pythonpkg/src/vector_conversion.cpp:325]: Good point

**Comment by Mytherin** [comment]
Python 2 failure seems to be related to bytes vs strings. Maybe just skip the newly added test on Python 2?

**Pull request merged**